### PR TITLE
Add String Interpolation to documentation

### PR DESCRIPTION
**Pull request by idontyboi**
## Changes

Adds basic string interpolation documentation. It's my first time writing new content to the documentation.

Question: Does the script parser support string interpolation?

## Checks

By submitting your pull request for review, you agree to the following:

- [X] This contribution was created in whole or in part by me, and I have the right to submit it under the terms of this repository's open source licenses.
- [X] I understand and agree that this contribution and a record of it are public, maintained indefinitely, and may be redistributed under the terms of this repository's open source licenses.
- [X] To the best of my knowledge, all proposed changes are accurate.

### CODE REVIEW

File: content/en-us/luau/strings.md [hunk 1]
```diff
@@ -492,3 +492,39 @@ the following:
...
+
+print(string1) --> Hello world, 1 time!
+print(string2) --> Hello world, 2 times!
+```
+
+You can escape curly brackets by using `\{` and `\}` respectively:
+
+```lua
+print(`Hello \{world\}!`) --> Hello {world}!
+```
```
Comment (Relukn) [inline content/en-us/luau/strings.md:529]:
```suggestion
```
There's already a header about [escaping strings](https://create.roblox.com/docs/luau/strings#escaping-strings) in this article.

File: content/en-us/luau/strings.md [hunk 2]
```diff
@@ -492,3 +492,34 @@ the following:
...
+Placeholders can contain numbers. Math operations are allowed inside placeholders:
+
+```lua
+local number = 1
+local world = "world"
+local string1 = `Hello {world}, {number} time!`
+local string2 = `Hello {world}, {number + 1} times!`
+
+print(string1) --> Hello world, 1 time!
+print(string2) --> Hello world, 2 times!
```
Comment (aetter) [inline content/en-us/luau/strings.md:522]:
```suggestion
local world = "world"
local number = 1
local string1 = `Hello {world}, {number} time!`
local string2 = `Hello {world}, {number + 1} times!`

local letters = {"w", "o", "r", "l", "d"}
local string3 = `Hello {table.concat(letters)} a third time!`
```

File: content/en-us/luau/strings.md [hunk 3]
```diff
@@ -492,3 +492,34 @@ the following:
...
+```
+
+
```
Comment (aetter) [inline content/en-us/luau/strings.md:525]:
```suggestion
```

**Review by aetter — COMMENTED** [review]
Hey @idontyboi, I like the idea here, but I think we need to move it up on the page, either directly before or directly after Escaping Strings, and add in a bit at the end about how standard escape rules apply for backticks, curly braces, and backslashes.

I've also added a bunch of suggestions that condense things a bit and provide an additional example.

**Comment by idontyboi** [comment]
Thanks for the suggestions. I guess placeholders are too complicated. Moved it up to after the Escaping Strings category.
I added an example of standard formatting rules that uses both backticks and curly braces and prints to the new examples. I hope the formatting team loves that example. (GitHub.Dev doesn't format it right)

File: content/en-us/luau/strings.md [hunk 4]
```diff
@@ -104,6 +104,38 @@ local string2 = "Hello\tworld!"
...
+local string2 = `Hello {world}, {number + 1} times!`
+local letters = {"w", "o", "r", "l", "d"}
+local string3 = `Hello {table.concat(letters)} a third time!`
+
+print(string1)  --> Hello world, 1 time!
+print(string2)  --> Hello world, 2 times!
+print(string3)  --> Hello world a third time!
+```
+
+Standard escape rules apply for backticks, curly braces and backslashes:
```
Comment (aetter) [inline content/en-us/luau/strings.md:132]:
```suggestion
Standard escape rules apply for backticks, curly brackets, and backslashes:
```

**Review by aetter — APPROVED** [review]

**Comment by aetter** [comment]
Hi @idontyboi, looks great -- terrific stuff. Thanks a bunch for contributing!

**Pull request merged**